homework/index.js

```diff
@@ -1,53 +1,143 @@
 'use strict';
 
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status >= 200 && xhr.status <= 299) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+  function fetchJSON(url) {
+    return new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
+        if (xhr.status >= 200 && xhr.status <= 299) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
   }
 
   function createAndAppend(name, parent, options = {}) {
     const elem = document.createElement(name);
     parent.appendChild(elem);
-    Object.entries(options).forEach(([key, value]) => {
+    Object.entries(options).forEach(([key, content]) => {
       if (key === 'text') {
-        elem.textContent = value;
+        elem.textContent = content;
       } else {
-        elem.setAttribute(key, value);
+        elem.setAttribute(key, content);
       }
     });
     return elem;
   }
 
+  function changeDateFormat(date) {
+    const timeFormat = new Date(date);
+    return timeFormat.toLocaleString();
+  }
+
+  function createTable(table, header, content) {
+    const row = createAndAppend('tr', table);
+    createAndAppend('th', row, { text: header, class: 'header' });
+    createAndAppend('td', row, { text: content, class: 'content' });
+    return row;
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const listItem = createAndAppend('li', ul, { class: 'listItem' });
+    const table = createAndAppend('table', listItem, { class: 'table' });
+    const topRow = createTable(table, 'Repository: ');
+    createAndAppend('a', topRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+    });
+    createTable(table, 'Description: ', repo.description);
+    createTable(table, 'Fork count: ', repo.forks);
+    createTable(table, 'Created on: ', changeDateFormat(repo.created_at));
+    createTable(table, 'Updated on: ', changeDateFormat(repo.updated_at));
+  }
+
+  function contributorsSection(url, parent) {
+    fetchJSON(url)
+      .then(repos => {
```

Comment:
Unfortunately, you are not telling the truth here. You are suggesting to the reader that you are fetching an array of repository objects, wherein truth you are fetching an array of contributor objects. This is misleading both for the readers of your code as for yourself.